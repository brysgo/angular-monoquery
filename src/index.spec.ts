import { HttpClientModule } from "@angular/common/http";
import { ApolloModule, Apollo } from "apollo-angular";
import { HttpLinkModule, HttpLink } from "apollo-angular-link-http";


import { TestBed } from "@angular/core/testing";

import gql from "graphql-tag";

describe("MonoQueryProvider", () => {
  let monoQueryProvider: MonoQueryProvider;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientModule, ApolloModule, HttpLinkModule],
      providers: [
        HttpClientModule,
        ApolloModule,
        HttpLinkModule,
        MonoQueryProvider
      ]
    });
    monoQueryProvider = TestBed.get(MonoQueryProvider);
  });

  it("should exist", () => {
    expect(monoQueryProvider).toBeDefined();
  });

  it("should use HttpClient", async () => {
    await monoQueryProvider.fetchData();
  });
});
