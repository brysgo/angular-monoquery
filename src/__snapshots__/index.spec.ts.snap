// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonoQueryModule is used to build a MonoProvider 1`] = `
<example-component
  monoProvider={[Function MonoProvider]}
>
  {"simpleFragment":{"hello":"world"}}
</example-component>
`;
